packages/webpack-cli/lib/groups/runHelp.js

```diff
@@ -90,48 +98,93 @@ const outputHelp = (args) => {
+            return allFlags
+                .filter((flag) => flag.negative)
+                .reduce((allFlags, flag) => {
+                    // Use available description for built-in negated flags
+                    const description = flag.negatedDescription ? flag.negatedDescription : `Negates ${flag.name}`;
+                    return [...allFlags, { name: `no-${flag.name}`, description, type: Boolean }];
+                }, []);
+        };
+
         const title = bold('⬡                     ') + underline('webpack') + bold('                     ⬡');
         const desc = 'The build tool for modern web applications';
         const websitelink = '         ' + underline('https://webpack.js.org');
         const usage = bold('Usage') + ': ' + '`' + green('webpack [...options] | <command>') + '`';
         const examples = bold('Example') + ': ' + '`' + green('webpack help --flag | <command>') + '`';
         const hh = `          ${title}\n\n${websitelink}\n\n${desc}\n\n${usage}\n${examples}`;
-        const output = commandLineUsage([
-            { content: hh, raw: true },
-            {
-                header: 'Available Commands',
-                content: commands.map((cmd) => {
-                    return { name: `${cmd.name} | ${cmd.alias}`, summary: cmd.description };
-                }),
-            },
-            {
-                header: 'Options',
-                optionList: flags
-                    .map((e) => {
-                        if (e.type.length > 1) {
-                            e.type = e.type[0];
-                        }
-
-                        // Here we replace special characters with chalk's escape
-                        // syntax (`\$&`) to avoid chalk trying to re-process our input.
-                        // This is needed because chalk supports a form of `{var}`
-                        // interpolation.
-                        e.description = e.description.replace(/[{}\\]/g, '\\$&');
-
-                        return e;
-                    })
-                    .concat(negatedFlags),
-            },
-        ]);
-
-        logger.raw(output);
+
+        if (args.includes('verbose') || args.includes('--help=verbose')) {
+            flagsToDisplay = [...flags, ...negatedFlags(flags)];
+
+            const headerAndCommands = commandLineUsage([
+                { content: hh, raw: true },
+                {
+                    header: 'Commands',
+                    content: commands.map((cmd) => {
+                        return { name: `${cmd.name} | ${cmd.alias}`, summary: cmd.description };
+                    }),
+                },
+                { header: 'Options', raw: true },
+            ]);
+
+            // print help-header & commands
+            logger.raw(headerAndCommands);
+            // print all options
+            for (const flag of flagsToDisplay) {
+                let flagType;
+
+                if (flag.multiple && !Array.isArray(flag.type)) {
```

Comment:
Still invalid, again, types can be multiple regardless of their type

Reply:
can you give example? I have covered the following cases - 

- `multiple: true, type: [Boolean, String] (Any combo)`  => `type = boolean, string[]`
- `multiple: flase, type: [Boolean, String] (Any combo)` => `type = boolean, string`
- `multiple: true, type: String` => `type = string[]`
- `multiple: flase, type: String` => `type = string`

Reply:
/cc @alexander-akait

Reply:
```
if (Array.isArray(flag.type)) {
                    const allowedTypes = flag.type.reduce((allTypes, type) => {
                        const currentType = flag.multiple ? `${type.name.toLowerCase()}[]` : type.name.toLowerCase();
                        return [...allTypes, currentType];
                    }, []);
                    flagType = allowedTypes.join(', ');
} else {
   flagType = flag.type.name.toLowerCase();

  if (flag.multiple) {
   flagType += `${flagType}[]`
  }
}
```

Pseudo code